compute_sdk/globus_compute_sdk/sdk/login_manager/tokenstore.py

```diff
@@ -36,11 +36,25 @@ def invalidate_old_config() -> None:
 
 
 def _ensure_funcx_dir() -> pathlib.Path:
-    dirname = _home() / ".funcx"
+    legacy_dirname = _home() / ".funcx"
+    dirname = _home() / ".globus_compute"
+
     try:
-        os.makedirs(dirname)
+        if os.path.exists(legacy_dirname) and os.path.isdir(legacy_dirname):
+            # If the legacy dir (.funcx) already exists, create a symlink
+            os.symlink(legacy_dirname, dirname, target_is_directory=True)
+
+        elif os.path.islink(dirname) and not os.path.exists(dirname):
+            # If the symlink is broken, remove it and create a fresh dir
+            os.unlink(dirname)
+            os.makedirs(dirname)
+
+        else:
+            # Otherwise, just create the dir
+            os.makedirs(dirname)
```

Comment:
What if .globus_compute exists but is a regular file?  I think it'll hit FileExistsError and we still assume it's fine, but it's not and will raise later.  This is existing behavior but maybe it's a chance to catch it and inform via a better message now that we're refactoring a bit.

Reply:
I'm ... agnostic here.  Your point is well taken, but I think this is very likely an edge case that "won't happen."  (FLW, I know.)  I hadn't gotten to this comment when I wrote the above suggestion, but I was intentionally ignoring this particular possibility.  My thinking then (and now!) is basically:

1. This is super unlikely
2. If, by golly, this _does_ occur, then the user will either
    - observe the error message and self-correct the issue, or
    - reach out (e.g., via Slack), and we'll see how wrong assumption of (1) is.

Reply:
Cool; as I said: "I'm ... agnostic here."